routes/socket/routes.js

```diff
@@ -106,10 +107,58 @@ module.exports = (modUserNames, editorUserNames, adminUserNames) => {
 			});
 		}
 
+		let isRestricted = !authenticated;
+
+		let checkingRestriction = false;
+		const checkRestriction = account => {
+			if (account) {
+				const parseVer = ver => {
+					let vals = ver.split('.');
+					vals.forEach((v, i) => (vals[i] = parseInt(v)));
+					return vals;
+				};
+				const firstVerNew = (v1, v2) => {
+					let pos = 0;
+					while (true) {

```

Comment:
`while true` loops should generally be avoided in favor of `forEach` or if need be vanilla for loops